app/services/remote_tagging_service.rb

```diff
@@ -46,7 +47,7 @@ def object_url
 
   def service_url
     match = self.class.remotes.detect { |item| item[:app_name] == @app_name && item[:object_type] == @object_type }
-    raise "No url found for app #{@app_name} object #{@object_type}" unless match
+    raise Exceptions::InvalidURLError.new("No url found for app #{@app_name} object #{@object_type}") unless match
```

Comment:
InvalidURLError is misleading to me. `Exceptions::UserError` is more appropriate because we cannot find a matched preconfigured URL based on user provided `app_name` and `object_type`. To users they do not know there is an internal URL involved.